src/app/application.cpp

```diff
@@ -173,6 +174,9 @@ Application::Application(int &argc, char **argv)
     SettingsStorage::initInstance();
     Preferences::initInstance();
 
+    if (!upgrade())
+        throw RuntimeError(u"Failed migration of old settings"_qs); // Not translatable. Translation isn't configured yet.
+
```

Comment:
FYI, this exception is uncaught in main(). I deliberately left it like this. Do you think I should catch it and display the message?
My rationale to leave it uncaught: AFAIK at this point we don't have a mainloop or application class. I doubt if a msgbox will work.

Reply:
>Do you think I should catch it and display the message?

Or should it be converted to an `std::exception` in order to have a better chance of the OS displaying something useful to the end-user?

Reply:
At least I would catch it and return EXIT_FAILURE like it was before.